user/user.go

```diff
@@ -51,17 +54,58 @@ func Create(response http.ResponseWriter, request *http.Request) {
 }
 
 // helper functions perform CRUD operations on user
-// func FindUserByID(response http.ResponseWriter, request *http.Request) {
-// 	user := &User{}
-// 	collectionName := "users"
-// 	objID, _ := primitive.ObjectIDFromHex(id)
-// 	collection := utils.GetCollection(collectionName)
-// 	res := collection.FindOne(ctx, bson.M{"_id": objID})
-// 	if err := res.Decode(user); err != nil {
-// 		return nil, err
-// 	}
-// 	return user, nil
-// }
+func FindUserByID(response http.ResponseWriter, request *http.Request) {
+	// Find a user by user ID
+	response.Header().Set("content-type", "application/json")
+
+	collectionName := "users"
+	userID := mux.Vars(request)["id"]
+	objID, _ := primitive.ObjectIDFromHex(userID)
```

Comment:
Please cater for a situation where by the supplied Id is a wrong one and primitive throws error. Currently with this, the application will get stuck when an invalid primitive is supplied.

Reply:
Resolved this a few minutes ago.